src/engine/ruleset/decoders/sonicwall/firewall-general.yml

```diff
@@ -0,0 +1,77 @@
+---
+name: decoder/sonicwall-firewall-general/0
+
+metadata:
+  module: firewall
+  title: Firewall logs decoder
+  description: Decoder for Linux firewall logs
+  author:
+    name: Wazuh Inc. info@wazuh.com
+    date: 2023-01-12
+  references:
+    - https://www.sonicwall.com/es-mx/support/technical-documentation
+
+sources:
+  - decoder/syslog/0
+  - decoder/queue-localfile/0
+
+check:
+  - event.original: +r_match/id\=.*?sn\=.*?time\=.*?fw
```

Comment:
I think a march with that regex is too generic to inherit from the syslog and localfile queues. Note that while it has the lazy quantifiers, this does not imply that it cannot have fields in between, since it is checked against `.`.
On the other hand, it would also be slow to test this regex for each character of all the logs that try to pass through this decoder, the resolution time of the regex will be proportional to the length of the log.
I can think of 2 possible ways to improve this:
- Force the regex to check only at the beginning of the string with ^, it would be something like: 
`\w{3} +?\d+? \d+:\d+:\d+ (ipv4|ipv6) id\=.*?sn\=.*?time\=.*?fw`
- Another option, to avoid regex, could be to add a use only parser with `<~log.timestamp/date/%b %d %T> <host.ip> <~log.message>` skipping the check, but I am not sure about this :thinking: . (This would imply fixing kvmap in this PR).

Reply:
The id and serial number, can they have spaces? if they can't, then don't use `.` and use the accepted characters (`alfa` or `\S`).